algebra/Atiyah.lean

```diff
@@ -0,0 +1,1695 @@
+import algebra.ring algebra.field data.set.basic order.zorn tactic.norm_num tactic.ring
+
+universes u v w
+
+-- page 1
+
+class zero_ring (α : Type u) [comm_ring α] : Prop :=
+(eq_zero : ∀ x:α, x = 0)
+
+instance zero_ring_of_zero_eq_one {α : Type u} [comm_ring α] : (0:α)=1 → zero_ring α
+| h := ⟨λ x, calc
+  x = x * 1   : by simp
+  ... = x * 0 : by rw h
+  ... = 0     : by simp⟩
+
+
+-- page 2
+
+class is_hom {α : Type u} {β : Type v} [comm_ring α] [comm_ring β] (f : α → β) : Prop :=
```

Comment:
this is in the root namespace, we will have a lot of `is_hom` or similar. I would prefer `is_ring_hom` or `ring.is_hom`. Also: would it make sense to generalize it from `comm_ring` to `ring`?

Reply:
Another idea: you use the `class` mechanism to introduce the hom structure. In the future we surely want a group morphism and `is_hom` and extension of group morphisms.